### PR TITLE
[Revolutionary] Add scenario search function to az interactive

### PR DESCRIPTION
**Pull request by wu5bocheng**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/interactive/README.md [hunk 1]
```diff
@@ -105,40 +114,78 @@ Only clears the appended suggestion when you restart the interactive shell
 
 The color option will be saved.
 
-## Cli Recommendation(Preview)
-2023.04.04
-####  **[Breaking Change]** Integrate the cli recommendation to make the completion ability more intelligent and provide the scenario completion.
+# Intelligent AZ Interactive(Revolutionary Change)
```
Comment (wu5bocheng) [inline src/interactive/README.md]:
```suggestion
# Intelligent AZ Interactive (Revolutionary Change)
```

File: src/interactive/README.md [hunk 2]
```diff
@@ -105,40 +114,78 @@ Only clears the appended suggestion when you restart the interactive shell
...
+The loading bar is an essential tool that improves the user experience by ensuring that all necessary components are loaded before usage, thereby reducing the risk of encountering errors and providing a smoother, more stable platform.
+
+Please refer to the following gif to see the loading bar in action:
+
+<div  align="center">    
+<img src="docs/loading_bar.gif" width = "400" height = "150" alt="loading_bar" align=center>
+</div>
+
+
+## Memory and Completion Mechanism for Param Value in Scenarios
```
Comment (wu5bocheng) [inline src/interactive/README.md]:
```suggestion
## Value Completion in Scenarios
```

File: src/interactive/azext_interactive/azclishell/app.py [hunk 3]
```diff
@@ -605,7 +607,48 @@ def scenario_repl(self, scenario):
...
+            time_spent_on_loading = time.time() - start_time
+            if time_spent_on_loading > prompt_timeout_limit:
+                print_styled_text([(Style.WARNING, 'Timeout to get search results.')])
+                break
+            if self.recommender.cur_thread.result:
+                break
+        if self.recommender.cur_thread.result is not None:
+            results = self.recommender.cur_thread.result
+            if len(results) == 0:
+                print_styled_text([(Style.WARNING, 'No search results.')])
```
Comment (wu5bocheng) [inline src/interactive/azext_interactive/azclishell/app.py]: give more user friendly hint

File: src/interactive/azext_interactive/azclishell/app.py [hunk 4]
```diff
@@ -651,9 +694,13 @@ def _special_cases(self, cmd, outside):
         elif cmd.startswith(SELECT_SYMBOL['example']):
             self.handle_scenario(cmd)
             continue_flag = True
+        elif cmd.strip().startswith(SELECT_SYMBOL['search']):
+            self.handle_search(cmd)
+            continue_flag = True
         elif SELECT_SYMBOL['example'] in cmd:
             cmd, continue_flag = self.handle_example(cmd, continue_flag)
             telemetry.track_ran_tutorial()
+            telemetry.track_ran_tutorial()
```
Comment (wu5bocheng) [inline src/interactive/azext_interactive/azclishell/app.py]:
```suggestion
```

File: src/interactive/azext_interactive/azclishell/recommendation.py [hunk 5]
```diff
@@ -250,7 +269,7 @@ def get_recommend_from_api(command_list, type, top_num=5, error_info=None):  # p
 
 
 def send_feedback(option_idx, latest_commands, processed_exception=None, recommends=None, accepted_recommend=None,
-                  api_version=None):
+                  api_version=None, request_type=1):
```
Comment (wu5bocheng) [inline src/interactive/azext_interactive/azclishell/recommendation.py]:
```suggestion
                  api_version=None, request_type=RecommendType.All):
```

File: src/interactive/azext_interactive/azclishell/recommendation.py [hunk 6]
```diff
@@ -298,7 +321,7 @@ def send_feedback(option_idx, latest_commands, processed_exception=None, recomme
     if accepted_recommend:
         feedback_data["accepted_recommend_source"] = accepted_recommend['source']
         feedback_data["accepted_recommend_type"] = accepted_recommend['type']
-        if accepted_recommend['type'] == RecommendType.Scenario:
+        if accepted_recommend['type'] == RecommendType.Scenario or accepted_recommend['type'] == RecommendType.Search:
```
Comment (zhoxing-ms) [inline src/interactive/azext_interactive/azclishell/recommendation.py]:
```suggestion
        if accepted_recommend['type'] in [RecommendType.Scenario, RecommendType.Search]:
```

File: src/interactive/azext_interactive/azclishell/scenario_suggest.py [hunk 7]
```diff
@@ -84,7 +84,7 @@ def get_suggestion(self, cli, buffer, document, value_storage_cache={}, auto_com
             unfinished = user_input.rsplit(' ', 1)[-1]
             # list of unused parameters in current sample
             unused_param = list(self.param_sample_value_map.keys())
-            completed_parts = user_input[-len(unfinished):].strip().split()
+            completed_parts = user_input.rsplit(' ', 1)[0].strip().split()
```
Comment (wu5bocheng) [inline src/interactive/azext_interactive/azclishell/scenario_suggest.py:87]: add comment

**Pull request merged**